packages/builder-vite/codegen-modern-iframe-script.ts

```diff
@@ -4,22 +4,49 @@ import { transformAbsPath } from './utils/transform-abs-path';
 import type { ExtendedOptions } from './types';
 
 export async function generateModernIframeScriptCode(options: ExtendedOptions) {
-  const { presets, configDir } = options;
+  const { presets, configDir, framework } = options;
 
   const previewOrConfigFile = loadPreviewOrConfigFile({ configDir });
   const presetEntries = await presets.apply('config', [], options);
   const configEntries = [...presetEntries, previewOrConfigFile]
     .filter(Boolean)
     .map((configEntry) => transformAbsPath(configEntry));
 
-  // noinspection UnnecessaryLocalVariableJS
+  const generateHMRHandler = (framework: string): string => {
+    // Web components are not compatible with HMR, so disable HMR, reload page instead.
+    if (framework === 'web-components') {
+      return `
+      if (import.meta.hot) {
+        import.meta.hot.accept((newModule) => {
+          const currentLocationHref = window.location.href;
```

Comment:
Can we change this to `import.meta.hot.decline()` ? https://vitejs.dev/guide/api-hmr.html#hot-decline

Reply:
👍 I did a little benchmarking, and using `decline` is indeed a little faster.  It was consistently taking 0.75 sec, vs 0.7 - 1.0 the other way I had it.